crypto.go

```diff
@@ -265,29 +265,34 @@ func deriveChildKey(parentKey []byte, chainCode []byte, index uint32) ([]byte, [
 	return digest[0:32], digest[32:]
 }
 
-func deriveLegacyChildKey(parentKey []byte, index int32) []byte {
-	in := make([]byte, 4)
-	k := 0
-	binary.LittleEndian.PutUint32(in, uint32(index))
-	password := make([]uint8, len(parentKey)+8)
-	for i, number := range parentKey {
-		password[i] = number
-	}
-
-	if index >= 0 {
-		for i := len(parentKey); i < len(parentKey)+4; i++ {
-			password[i] = 0
+func deriveLegacyChildKey(parentKey []byte, index int64) []byte {
+	in := make([]uint8, 8)
+	if index == int64(0xffffffffff) {
+		in[3] = uint8(0xff)
+		for i := 4; i < len(in); i++ {
+			in[i] = uint8(uint64(index) >> 32)
```

Comment:
A bit confused on what this is trying to do. if `index` is `0xff_ffffffff` then you can just set `in` to be `{0, 0, 0, 0xff, 0xff, 0xff, 0xff, 0xff}`?